src/memray/reporters/assets/temporal_flamegraph.js

```diff
@@ -52,7 +52,7 @@ function packedDataToTree(packedData, rangeStart, rangeEnd) {
   }
 
   const hwms = packedData.high_water_mark_by_snapshot;
-  if (hwms) {
+  if (hwms && hwms.length > 0 && memory_records.length > 0) {
```

Comment:
This fix isn't correct. The `else` block of this `if` is for temporal leaks mode, rather than temporal high water mark mode. We don't want to go into that if `hwms` is set, even if it's set to an empty array.